lib/graphql/auth-guards.ts

```diff
@@ -0,0 +1,16 @@
+import { Role } from '@prisma/client';
+import { Resolver } from './resolvers';
+
+export const authenticate =
+  (resolver: Resolver, authorizedRoles: Array<Role>): Resolver =>
+  (root, args, context, info) => {
+    const userRole = context.session?.user?.role;
```

Comment:
build is failing with error `'role' does not exist` on the user. Kinda confused by this since i thought that get_session would return what is included in the next_auth session callback.
https://github.com/uwblueprint/richmond-centre-for-disability/blob/06c6b000f9505440d08e18a26464cbb960d5e3c8/pages/api/auth/%5B...nextauth%5D.ts#L65-L71

Alternative would be to make a db call with the session email in context.ts and get the user that way. Anyone have any thoughts?

Reply:
Hmm, looks like it's an issue with GraphQL playground configuration. This SO post has a solution https://stackoverflow.com/a/66308267.

Reply:
tried this but didn't do the trick. After a bunch of looking around i found that we have to override the default next-auth types if we change what they output https://next-auth.js.org/getting-started/typescript#module-augmentation